addons/knobs/src/components/types/Select.tsx

```diff
@@ -30,15 +30,23 @@ const SelectType: FunctionComponent<SelectTypeProps> & {
   deserialize: typeof deserialize;
 } = ({ knob, onChange }) => {
   const { options } = knob;
-  const entries = Array.isArray(options)
-    ? options.reduce<Record<PropertyKey, SelectTypeKnobValue>>((acc, k) => ({ ...acc, [k]: k }), {})
-    : (options as Record<PropertyKey, SelectTypeKnobValue>);
 
-  const selectedKey = Object.keys(entries).find(k => {
-    if (Array.isArray(knob.value)) {
-      return JSON.stringify(entries[k]) === JSON.stringify(knob.value);
+  const callbackReduceArrayOptions = (acc: any, option: any, i: number) => {
+    if (typeof option !== 'object') return { ...acc, [option]: option };
+    const label = option.label || option.key || i;
+    return { ...acc, [label]: option };
+  };
```

Comment:
Does this mean that my objects need to have a `label` or `key` attribute to have a display value? That's a little funky. The data passed to the react component shouldn't need to include storybook specific labels.

Something like this would keep the storybook integration separate from the data being passed to the component:
```
const arrayOfObjects = [
  {
    label: 'Sparky', // storybook specific label for the option
    data: { // passed to the react component
        dogName: 'Sparky',
        dogParent: 'Matthew',
        location: 'Austin',
    }
  },
  {
    label: 'Juniper',
    data: {
       dogName: 'Juniper',
       dogParent: 'Joshua',
       location: 'Austin',
    }
  },
];
```

Reply:
> Does this mean that my objects need to have a label or key attribute to have a display value?

1. there's a fallback `const label = option.label || option.key || i;` to the index (i.e. `i`). So it'll always work (label might just not look very nice) if a user chooses to not pass a label or key.

2. Also, the `label` or `key` would just need to be passed in on the storybook level (react logic does NOT need to have a label). In fact: you show a great example of how to do this in your code snippet!
```
// using your `arrayOfObjects` example
const dogInfo = select(<label>, arrayOfObjects).data // notice we use.data`
// pass dogInfo to react object.
```